addons/account/models/sequence_mixin.py

```diff
@@ -20,6 +20,38 @@ class SequenceMixin(models.AbstractModel):
     _sequence_yearly_regex = r'^(?P<prefix1>.*?)(?P<year>\d{4})(?P<prefix2>\D+?)(?P<seq>\d*)(?P<suffix>\D*?)$'
     _sequence_fixed_regex = r'^(?P<prefix1>.*?)(?P<seq>\d*)(?P<suffix>\D*?)$'
 
+    sequence_prefix = fields.Char(compute='_compute_split_sequence', store=True)
+    sequence_number = fields.Integer(compute='_compute_split_sequence', store=True)
+
+    def __init__(self, pool, cr):
+        api.depends(self._sequence_field)(type(self)._compute_split_sequence)
+        return super().__init__(pool, cr)
+
+    def init(self):
+        """ change index on partner_id to a multi-column index on (partner_id, ref), the new index will behave in the
+            same way when we search on partner_id, with the addition of being optimal when having a query that will
+            search on partner_id and ref at the same time (which is the case when we open the bank reconciliation widget)
+        """
+        if not self._abstract:
+            index_name = self._table + '_sequence_index'
+            self.env.cr.execute('SELECT indexname FROM pg_indexes WHERE indexname = %s', (index_name,))
+            if not self.env.cr.fetchone():
+                self.env.cr.execute("""
+                CREATE INDEX {index_name} ON {table}
+                    (journal_id, sequence_prefix desc, sequence_number desc, date, {field});
+                """.format(
+                    index_name=index_name,
+                    table=self._table,
+                    field=self._sequence_field,
+                ))
+
+    def _compute_split_sequence(self):
+        for record in self:
+            matching = re.match(record._sequence_fixed_regex, record[record._sequence_field] or '')
+            record.sequence_prefix = matching.group('prefix1')
+            record.sequence_number = int(matching.group('seq') or 0)
+        self.flush(['sequence_number', 'sequence_prefix'])
```

Comment:
why is a flush needed?

Reply:
Because we use those fields in the query. If we don't flush here, it will only be reflected in the database later (possibly at the end of all the computes), but we need it in `_compute_name` of `account.move` for instance, before the end of all computes.

Reply:
Even if the flush is needed later, it's not the role of the compute function to flush.